web/src/api/images/get_unverified_images.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as getUnverifiedImagesSDK } from "@/api/images/graphql/getUnverifiedImages.generated";
+import { Session, getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+import { logger } from "@/lib/logger";
+
+export type ImageGetAllUnverifiedImagesResponse = {
+  urls?: {
+    logo_img_url?: string;
+    hero_image_url?: string;
+    showcase_img_urls?: string[];
+  };
+  success?: boolean;
+};
+
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+});
+
+export type ImageGetAllUnverifiedImagesBody = yup.InferType<typeof schema>;
+// This endpoint takes in an App ID and returns all available unverified images for that app.
```

Comment:
Actually when you upload an image but don't save yet this won't be able to pull it. We also don't know what extension to manually get. Feel like at that point it might be easier to separate the logic. Would require us to then have get_all_unverified generate signed URLs even if an image doesn't exist there.